gokart/utils.py

```diff
@@ -7,3 +11,48 @@ def add_config(file_path: str):
     _, ext = os.path.splitext(file_path)
     luigi.configuration.core.parser = ext
     assert luigi.configuration.add_config_path(file_path)
+
+
+T = TypeVar('T')
+if sys.version_info >= (3, 10):
+    from typing import TypeAlias
+
+    FlattableItems: TypeAlias = T | Iterable['FlattableItems[T]'] | dict[str, 'FlattableItems[T]']
+else:
+    from typing import Union
+
+    FlattableItems = Union[T, Iterable['FlattableItems[T]'], dict[str, 'FlattableItems[T]']]
+
+
+def flatten(targets: FlattableItems[T]) -> list[T]:
```

Comment:
@kitagry plz note this is slightly changed copy of luigi.task.flatten https://github.com/spotify/luigi/blob/367edc2e3a099b8a0c2d15b1676269e33ad06117/luigi/task.py#L958

Reply:
How about testing working the same way as luigi.task.flatten?



``
self.assertEqual(sorted(luigi.task.flatten({'a': 'foo', 'b': 'bar'})), sorted(gokart.utils.flatten({'a': 'foo', 'b': 'bar'})))
``

Reply:
add note [57511ea](https://github.com/m3dev/gokart/pull/362/commits/57511ea3b4e970e854a69802824a192dd3ca4295)

Reply:
I added test. (I think it is better to use concrete values instead of comparing to `luigi.task.flatten`)